### PR TITLE
Fuzzing MVP

### PR DESCRIPTION
**Pull request by mingxguo27**
MVP of OpenSK HID fuzzing with dependencies and linking errors fixed.

### CODE REVIEW

File: fuzz/fuzz_targets/fuzz_target_split_assemble.rs
Comment (kaczmarczyck) [inline fuzz/fuzz_targets/fuzz_target_split_assemble.rs:51]: Copied from the last PR: Consider `split_last` here?

File: src/lib.rs
Comment (kaczmarczyck) [inline src/lib.rs:25]: When you run 

> ./deploy.py --opensk --board=nrf52840dk

you get

> error: unused `#[macro_use]` import

(You also see those with `./run_desktop_tests.sh`.)

**Review by kaczmarczyck — APPROVED** [review]
Let's wait for one more reviewer to do a pass.

File: fuzz/Cargo.toml
Comment (ia0) [inline fuzz/Cargo.toml:1]: Was the newline automatically generated? I think it's safe to remove.

File: fuzz/Cargo.toml
Comment (ia0) [inline fuzz/Cargo.toml:17]: Style: Missing space before closing brace.

File: fuzz/fuzz_targets/fuzz_target_split_assemble.rs
Comment (ia0) [inline fuzz/fuzz_targets/fuzz_target_split_assemble.rs:18]: The function seems to have redundant information. In Rust, a reference to a slice is a "fat pointer", i.e. it contains both a pointer to the slice and the length of the slice. So here, the `len` argument is redundant with `data`. You can replace all usages of `len` with `data.len()` in this function and remove the argument.

File: fuzz/fuzz_targets/fuzz_target_split_assemble.rs
Comment (ia0) [inline fuzz/fuzz_targets/fuzz_target_split_assemble.rs:42]: This is probably just style, but it removes a bit of the clutter. What about `let mut message = raw_to_message(data);` and then `message.clone()` and `message.cmd &= !PACKET_TYPE_MASK`?

File: third_party/lang-items/Cargo.toml
Comment (ia0) [inline third_party/lang-items/Cargo.toml:12]: I guess that's the solution to the problem we faced.

File: fuzz/fuzz_targets/fuzz_target_split_assemble.rs
Comment (mingxguo27) [reply fuzz/fuzz_targets/fuzz_target_split_assemble.rs:42]: Ahh I can't believe it :D I'm too dumb. I had to break it down into parts because I got an error saying Message didn't derive some trait, probably I tried .copy() instead of .clone() first.

File: third_party/lang-items/Cargo.toml
Comment (mingxguo27) [reply third_party/lang-items/Cargo.toml:12]: Yes :')

**Review by ia0 — DISMISSED** [review]

**Review by kaczmarczyck — DISMISSED** [review]

**Review by kaczmarczyck — DISMISSED** [review]

**Comment by kaczmarczyck** [comment]
@jmichelp The binary seems to not be reproducible anymore. Is this related to this pull request? I'd propose to squash and merge on this one, and investigate later?

**Comment by jmichelp** [comment]
Should be fixed by #153 
Please rebase

**Review by kaczmarczyck — DISMISSED** [review]
Let's give @gendx a chance to take a look next week. Otherwise LGTM.

File: src/ctap/storage.rs
Comment (gendx) [inline src/ctap/storage.rs:22]: Looks good to me, although I still don't fully understand why the import path changes. But we can discuss splitting away the `embedded_flash` into its own crate as a separate pull request.

File: src/lib.rs
Comment (gendx) [inline src/lib.rs:23]: Looks good, although we won't need these anymore when migrating to Rust 2018 edition (#118).

File: third_party/lang-items/Cargo.toml
Comment (gendx) [inline third_party/lang-items/Cargo.toml:12]: If we remove this feature, how is the allocator initialized in the real OpenSK application? Have you tried running OpenSK with this pull request?

**Review by gendx — CHANGES_REQUESTED** [review]

**Comment by gendx** [comment]
I'm concerned that removing the `alloc_init` feature from libtock-rs core breaks OpenSK, as no allocator is initialized.

**Review by gendx — COMMENTED** [review]
Now that #156 is merged, the changes in `src/` and `third_party/` shouldn't be needed anymore, and the `fuzzing` feature can be removed. The `std` and `ram_storage` features should be sufficient.

**Note:** `cargo fuzz` requires `cargo install cargo-fuzz`. This should be added to either `setup.sh` or a `fuzz-setup.sh` script.

File: third_party/lang-items/Cargo.toml
Comment (gendx) [reply third_party/lang-items/Cargo.toml:12]: Resolved by #156.

File: src/lib.rs
Comment (gendx) [reply src/lib.rs:23]: Resolved by #156.

File: src/ctap/storage.rs
Comment (gendx) [reply src/ctap/storage.rs:22]: Resolved by #156.

**Review by kaczmarczyck — DISMISSED** [review]

**Review by gendx — COMMENTED** [review]
It would be great to add a GitHub workflow that does `cargo fuzz build` to make sure this doesn't break in the future.

**Review by gendx — APPROVED** [review]

**Comment by gendx** [comment]
No questions left from me. I'll leave it to @kaczmarczyck to merge this pull request.

**Pull request merged**
